File: DemoServer/AdditionalLanguages/python/queries/projectingUsingFunctions/projectingUsingFunctions.py

#region Demo
class EmployeeDetails:
    def __init__(self, full_name: str, title: str):
        self.full_name = full_name
        self.title = title
#endregion

def run(self, run_params=None):
    #region Demo
    with self.document_store_holder.store().open_session() as session:
        
        raw_query_string = (
            #region Step_1
            "declare function output(employee) {"
            "   var formatName  = function(employee) { return 'FullName: ' + employee.FirstName + ' ' + employee.LastName; };"
            "   var formatTitle = function(employee) { return 'Title: ' + employee.Title };"
            "   return { title : formatTitle(employee), full_name : formatName(employee) };"
            "}"
            #endregion
            #region Step_2
            "from Employees as employee select output(employee)"
            #endregion
        )

        #region Step_3
        projected_query_with_functions = session.advanced.raw_query(
            raw_query_string, EmployeeDetails
        )
        #endregion

        #region Step_4
        projected_results = list(projected_query_with_functions)
        #endregion
    #endregion

    return projected_results

File: DemoServer/AdditionalLanguages/php/queries/projectingUsingFunctions/ProjectingUsingFunctions.php
<?php

namespace RavenDB\Demo\queries\projectingUsingFunctions;

//region Usings
use Symfony\Component\Serializer\Annotation\SerializedName;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;

//region Demo
class EmployeeDetails
{
    #[SerializedName("FullName")]
    public ?string $fullName = null;
    #[SerializedName("Title")]
    public ?string $title = null;
}
//endregion

class ProjectingUsingFunctions
{
    public function __invoke(): array
    {
        $projectedResults = [];
        //region Demo
        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            $rawQueryString =
                //region Step_1
                "declare function output(employee) {" .
                "   var formatName  = function(employee) { return 'FullName: ' + employee.FirstName + ' ' + employee.LastName; };" .
                "   var formatTitle = function(employee) { return 'Title: ' + employee.Title };" .
                "   return { Title : formatTitle(employee), FullName : formatName(employee) };" .
                "}".
                //endregion
                //region Step_2
                "from Employees as employee select output(employee)";
                //endregion

            //region Step_3
            $projectedQueryWithFunctions = $session->advanced()
                ->rawQuery(EmployeeDetails::class, $rawQueryString);
            //endregion

            //region Step_4
            $projectedResults = $projectedQueryWithFunctions->toList();
            //endregion
        //endregion
        
        } finally {
            $session->close();
        }
        
        return $projectedResults;
    }
}
